packages/desktop-client/src/components/reports/reports/CustomReport.jsx

```diff
@@ -293,9 +331,14 @@ export function CustomReport() {
               <AppliedFilters
                 filters={filters}
                 onUpdate={onUpdateFilter}
-                onDelete={onDeleteFilter}
+                onDelete={filter =>
+                  onChangeAppliedFilter(filter, onDeleteFilter)
+                }
                 conditionsOp={conditionsOp}
-                onCondOpChange={onCondOpChange}
+                onCondOpChange={filter =>
+                  onChangeAppliedFilter(filter, onCondOpChange)
+                }
+                onUpdateChange={onReportChange}
```

Comment:
Should also get in on the onApplyFilter of the toptoolbar as well. As currently this is only firing if the filters are changed from the applied filters. Not if they get added from the topbar.